File: tests/NewArgs.cs
namespace Demo {
    public partial class Foo {
        public static void Make() {
            System.Collections.Generic.List<int> t;
            t = new System.Collections.Generic.List<int>(5);
        }
        public static void Named() {
            Demo.Bar v;
            v = new Demo.Bar(1, "a");
        }
    }
}

File: tests/NewArgs.pas
namespace Demo;

type
  Foo = public class
  public
    class method Make;
    class method Named;
  end;

implementation

class method Foo.Make;
var t : System.Collections.Generic.List<Integer>;
begin
  t := new System.Collections.Generic.List<Integer>(5);
end;

class method Foo.Named;
var v : Demo.Bar;
begin
  v := new Demo.Bar(value := 1, name := 'a');
end;

end.
